python/cudf/cudf/core/column/column.py

```diff
@@ -1948,8 +1948,23 @@ def as_column(
                 raise TypeError(
                     f"Cannot convert a {inferred_dtype} of object type"
                 )
-            elif nan_as_null is False and (
-                pd.isna(arbitrary).any()
+            elif inferred_dtype == "boolean":
+                if cudf.get_option("mode.pandas_compatible"):
+                    if not (
+                        dtype is not None
+                        and dtype == np.dtype("bool")
```

Comment:
The second condition should imply the first, so we probably don't need both.
```suggestion
                        dtype == np.dtype("bool")
```